src/views/Viewer.vue

```diff
@@ -373,6 +365,13 @@ export default {
 				: mime
 		},
 
+		getPath(fileInfo) {
+			if (fileInfo.hasPreview) {
+				return generateUrl(`/core/preview?fileId=${fileInfo.id}&x=${window.outerWidth}&y=${window.outerHeight}&a=true`)
```

Comment:
Might be an idea to respect already generated sizes?
Does the endpoint generate exactly the size requested or does it choose the closest?
Like in rullzers app: https://github.com/rullzer/previewgenerator, previews are generated in sizes of multiples of 32.

Reply:
It will automatically.
The server will return the closest power of 2 :)

Reply:
@skjnldsv That means it could create a lot of different sizes then. What happens if we resize the browser window? Does it update the url with new sizes? It could quickly get out of hand I think.

At work we have 300k raw image files and with the preview generator app I created millions (!) of previews filling up the filecache table and making rsync backups of the file system super slow and making everything in nextcloud slow. It would be good if we used a srcset here and limited it to only a few different sizes.

Reply:
@cowai no, the server always generate a specific array of sizes (according to a power of 2). So me requesting any size will not let the server generate crazy odd sizes.
It will just return the already generated closest size of the values I requested :)